Agda fails to compile

```
    Configuring Agda-2.6.1.2...
    Preprocessing library for Agda-2.6.1.2..
    Building library for Agda-2.6.1.2..
    [ 33 of 369] Compiling Agda.Utils.Maybe
    
    /var/stackage/work/unpack-dir/unpacked/Agda-2.6.1.2-baf0581cd7bf2854805c8da461c5218d478db0699218f7857bd728f074c14a42/src/full/Agda/Utils/Maybe.hs:13:1: error:
        Could not load module ‘Control.Monad.Trans.Maybe’
        It is a member of the hidden package ‘transformers-0.5.6.2’.
        Perhaps you need to add ‘transformers’ to the build-depends in your .cabal file.
        It is a member of the hidden package ‘rerebase-1.6.1’.
        Perhaps you need to add ‘rerebase’ to the build-depends in your .cabal file.
        Use -v (or `:set -v` in ghci) to see a list of the files searched for.
       |
    13 | import Control.Monad.Trans.Maybe
       | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

CC @andreasabel 

This started failing after #5824 which effectively enabled `Agda`.

--- a/build-constraints.yaml
+++ b/build-constraints.yaml
@@ -25,7 +25,7 @@ packages:
     "Andreas Abel <andreas.abel@gu.se> @andreasabel":
         - BNFC
         - STMonadTrans
-        - Agda < 0  # https://github.com/commercialhaskell/stackage/issues/5826
+        - Agda
         - agda2lagda
         - ListLike
         - haskell-src

